src/client/app/pages/SearchPage/SearchPage.jsx

```diff
@@ -1,69 +1,80 @@
 import React from 'react';
+import { connect } from 'react-redux'
+import { bindActionCreators } from 'redux'
 
 import Page from '../../components/Page'
 import SearchBar from './SearchBar/SearchBar'
 import SearchResultsInfo from './SearchResultsInfo/SearchResultsInfo'
 import SearchResults from './SearchResults/SearchResults'
-import movies from '../../test-data/dataMovies'
+import * as pageActions from './data/SearchPageActions'
+import {SORTS, SOURCES} from './data/consts'
 
-const sorts = [
-  {value: 'rating', title: 'rating'},
-  {value: 'releaseYear', title: 'release year'}
-];
-
-const sources = [
-  {value: 'title', title: 'title'},
-  {value: 'director', title: 'director'}
-];
-
-export default class SearchPage extends React.Component {
+class SearchPage extends React.Component {
   constructor (props) {
     super(props);
 
-    this.state = {
-      sortBy: sorts[0].value,
-      source: sources[0].value
-    };
-
     this.changeSort = this.changeSort.bind(this);
     this.changeSource = this.changeSource.bind(this);
-    this.search = this.search.bind(this);
+    this.changeQuery = this.changeQuery.bind(this);
+
+  }
+
+  componentWillMount () {
+    const query = this.props.match.params.searchQuery || '';
+    this.changeQuery(query);
   }
 
   changeSort (sortValue) {
-    this.setState({
-      sortBy: sortValue
-    });
+    this.props.pageActions.changeSort(sortValue);
+    this.props.pageActions.sort(this.props.page.movies, sortValue);
   }
 
   changeSource (sourceValue) {
-    this.setState({
-      source: sourceValue
-    });
+    this.props.pageActions.changeSource(sourceValue);
   }
 
-  search ({ searchQuery, filterBy }) {
-    // TO DO
+  changeQuery (query) {
+    this.props.pageActions.searchMovies(query, this.props.page.sortBy);
+
+    const searchUrl = '/search/' + query;
```

Comment:
It's better to define such paths and routes into a separate config files and import them here